bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Observation.scala

```diff
@@ -1,36 +1,43 @@
 package edu.gemini.model.p1.immutable
 
+import edu.gemini.model.p1.overheads.Overheads
 import edu.gemini.model.p1.{mutable => M}
+
 import scalaz._
 import Scalaz._
 
 object Observation {
 
-  // Some lenses
+  // Some lenses.
   val blueprint:Lens[Observation, Option[BlueprintBase]] = Lens.lensu((a, b) => a.copy(blueprint = b), _.blueprint)
   val condition:Lens[Observation, Option[Condition]] = Lens.lensu((a, b) => a.copy(condition = b), _.condition)
   val target:Lens[Observation, Option[Target]] = Lens.lensu((a, b) => a.copy(target = b), _.target)
   val meta:Lens[Observation, Option[ObservationMeta]] = Lens.lensu((a, b) => a.copy(meta = b), _.meta)
-  val time:Lens[Observation, Option[TimeAmount]] = Lens.lensu((a, b) => a.copy(time = b), _.time)
+  val intTime:Lens[Observation, Option[TimeAmount]] = Lens.lensu((a, b) => a.copy(intTime = b), _.intTime)
 
   def apply(blueprint:Option[BlueprintBase],
             condition:Option[Condition],
             target:Option[Target],
             band:Band,
-            time:Option[TimeAmount]) = new Observation(blueprint, condition, target, time, band)
+            intTime:Option[TimeAmount],
+            progTime:Option[TimeAmount] = None) = new Observation(blueprint, condition, target, intTime, progTime, band)
 
   def apply(m:M.Observation) = new Observation(m)
 
-  def unapply(o:Observation) = Some((o.blueprint, o.condition, o.target, o.time, o.band))
+  def unapply(o:Observation) = Some((o.blueprint, o.condition, o.target, o.intTime, o.calculatedTimes, o.band))
 
-  val empty = new Observation(None, None, None, None, M.Band.BAND_1_2)
+  val empty = new Observation(None, None, None, None, None, M.Band.BAND_1_2)
 
 }
 
+// REL-2985: It is unfortunate that we have to pass progTime here, but it is necessary for the migration to 2017B,
+// since the old "time" user-specified parameter is now progTime, and we have changed to have progTime now calculated
+// as a function of parameter intTime: thus, if progTime is defined, we must calculate intTime from it.
```

Comment:
`progTime` would be ignored.

I suppose we might be able to somehow separate this into two: one that accepts a `progTime`, calculates the `intTime`, and then creates the object based on this. I'd have to investigate a little more how that would figure into the mutable to immutable conversion when reading.